arm/cores/wiring_analog.h

```diff
@@ -40,47 +40,108 @@ extern "C" {
     extern void analogReference( uint8_t ulMode ) ;
 
     /*
-     * \brief Writes an analog value (PWM wave) to a pin.
+     * \brief Writes an analogue value to a DAC or PWM wave to a pin.
+     *        DAC is straight write to DAC (if present on that pin)
+     *
+     *        PWM depends on analogWriteResolution()
+     *        Effect of value is the duty cycle for PWM output to be HIGH
+     *        Valid values are
+     *                   Write resolution (bits)
+     *        Value      8      10      12      16
+     *        OFF        0      0       0       0
+     *        ON always  255    1023    4095    65535
+     *
+     *        Values in between these values vary the duty cycle
      *
      * \param pin
-     * \param value
+     * \param value must be 0 to max value for ADC WRITE resolution
+     *
+     * \return 0 = success, -1 = invalid value, -2 = wrong pin
      */
-    extern void analogWrite( uint8_t pin, uint16_t value ) ;
+    extern int16_t analogWrite( uint8_t pin, uint16_t value ) ;
```

Comment:
The Arduino Reference Language should be not be modified. API can be extended with new functionalities, but API compatibility with official boards must be preserved.